x/auth/ante.go

```diff
@@ -74,6 +74,9 @@ func NewAnteHandler(accountMapper sdk.AccountMapper) sdk.AnteHandler {
 				// TODO: min fee
 				if !fee.Amount.IsZero() {
 					signerAcc, res = deductFees(signerAcc, fee)
+					pool := accountMapper.GetFeePool(ctx)
```

Comment:
I'm not a fan of extending accountMapper in this way - I reckon we should keep accountMapper as is and  have AnteHandler take a new type of keeper (which can be defined in `/types`) which the stake module will define. The AnteHandler can have an additional function input say "FeeHandler" has a method `HandleFees(ctx, fee coin.Coins)` which actually performs the movement of the fees. `HandleFees` could be called right here

Reply:
But yeah then the reserve tax can be implemented as a part of `HandleFees` too

Reply:
The nice part about this implementation is that it's cheap (one read, one write) and complex fee redistribution can be performed later in a batch - but I'll implement that option in another PR, then we can see which we like best.

Reply:
yeah I mean it's actually not cheap because now we need to read write to the pool and then later read from the pool to perform the distribution instead of just tackling simultaneously

Reply:
Hmm, according to https://cosmos.network/whitepaper fees are redistributed every `ValidatorPayoutPeriod` (default 1 hour), so it seems like we don't need to do that every block? 

Even if we did redistribute every block instead, any implementation not writing to intermediary state would have to redistribute every transaction, which seems like it would be very inefficient (read/write all the bonded stakeholder bond values and accounts)?

Reply:
@rigelrozanski what do you mean by "digest"?

Reply:
apologies for the vague language - yeah digest = process the fee pool for the the proposer as outlined in the spec - https://github.com/cosmos/cosmos-sdk/blob/master/docs/spec/staking/old/spec2.md#fee-calculations

Also, due to technical reasons we must vear from the whitepaper proposed hourly distribution. Talked with Jae about this before - is OK. But yeah digest doesn't mean we're "distributing" to the validators we're calculating how the "feepool shares" ought to be modified for the proposer validator - this calculation is dependant on the fees collected in each block/digest. So yeah, the staking fee calculation actually needs access to what each incoming fee is - if it only has access to the pool in this current implementation, it would actually need to keep a second copy of the pool to calculate the difference, which is totally nuts when we could just do it all at once